src/vivarium_gates_mncnh/data/hemoglobin_effects/hgb_lbwsg_effect_generation.py

```diff
@@ -0,0 +1,367 @@
+    return results
+
+
+def scale_lbwsg_to_iv_iron(lbwsg_shifts, draw):
+    """Scales hemoglobin effects on GA and BW (relative to the hemoglobin TMREL) to the effect size of IV iron"""
+
+    # TODO: read in IV iron effect size/threshold from the repo/artifact here instead of hard-coding it
+    iv_iron_md = 23  # change in hemoglobin exposure associated with IV iron
+    iv_iron_threshold = 100  # maximum hemoglobin exposure level (g/L) eligible for IV iron
+
+    lbwsg_shifts["exposure"] = lbwsg_shifts["exposure"]
+    exposure_levels = lbwsg_shifts.exposure.unique().tolist()
+
+    # round to the nearest exposure increment
+    exposure_increment = round(exposure_levels[1] - exposure_levels[0], 4)
+    iv_iron_md = round(iv_iron_md / exposure_increment, 0) * exposure_increment
+
+    ga_shifts = lbwsg_shifts.loc[lbwsg_shifts.outcome == "gestational_age"]
+    ga_alt = ga_shifts.copy()
+    ga_alt["exposure"] = (ga_alt["exposure"] - iv_iron_md).round(
+        2
+    )  # this is the least amount of rounding that we can get away with
+    ga_shifts["exposure"] = ga_shifts["exposure"].round(2)
+    # merge with "left" method so that we keep all original exposure levels, but not the ones that have the IV iron MD subtracted
+    ga_shifts = ga_shifts.merge(
+        ga_alt,
+        on=["location", "sex", "draw", "exposure", "outcome"],
+        suffixes=("", "_alt"),
+        how="left",
+    )
+    ga_shifts["iv_iron_shift"] = ga_shifts["value_alt"] - ga_shifts["value"]
+
+    bw_shifts = lbwsg_shifts.loc[lbwsg_shifts.outcome == "birth_weight"]
+    bw_alt = bw_shifts.copy()
+    bw_alt["exposure"] = (bw_alt["exposure"] - iv_iron_md).round(2)
+    bw_shifts["exposure"] = bw_shifts["exposure"].round(2)
+    bw_shifts = bw_shifts.merge(
+        bw_alt,
+        on=["location", "sex", "draw", "exposure", "outcome"],
+        suffixes=("", "_alt"),
+        how="left",
+    )
+    bw_shifts["iv_iron_shift"] = bw_shifts["value_alt"] - bw_shifts["value"]
+
+    iv_iron_lbwsg_shifts = pd.concat([ga_shifts, bw_shifts], ignore_index=True)
+    iv_iron_lbwsg_shifts = iv_iron_lbwsg_shifts.drop(columns=["value_alt", "value"]).rename(
+        columns={"iv_iron_shift": "value"}
+    )
+    iv_iron_lbwsg_shifts["value"] = np.where(
+        iv_iron_lbwsg_shifts.exposure > iv_iron_threshold,
+        np.nan,
+        iv_iron_lbwsg_shifts["value"],
+    )  # missing values for exposure levels
+
+    iv_iron_lbwsg_shifts.sort_values(by=["location", "sex", "draw", "outcome", "exposure"])
+    return iv_iron_lbwsg_shifts
+
+
+def calculate_and_save_lbwsg_shifts(results_directory, draw):
+    assert os.path.exists(
+        results_directory + "lbwsg_shifts/"
+    ), f"Results directory {results_directory + 'lbwsg_shifts/'} does not exist"
+    lbwsg_shifts = get_lbwsg_shifts(draw)
+    lbwsg_shifts.to_csv(results_directory + "lbwsg_shifts/" + draw + ".csv", index=False)
+
+
+def calculate_and_save_iv_iron_lbwsg_shifts(results_directory, draw):
```

Comment:
This will be a point to get input from engineers: how should this should ultimately work? Should we make this an executable script that we can run with `python hgb_lbwsg_effect_generation.py`? I know you also mentioned including a notebook that will call these functions, so maybe that is enough.

Reply:
The code here is expected to take a long time to run if I recall correctly? It ultimately doesn't matter so the script can either be an executable script or generated via a notebook but make  sure you document how to run it well enough that someone who has never run it before could just sit down, read the documentation, and then generate the data.

I will say for other code that takes a short amount of time to run, it should just be part of the artifact building code in loader.py or called from any of the functions in loader.py.

Reply:
Planning to follow-up on this issue in a subsequent PR, so marking as resolved for now.